sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_response.py

```diff
@@ -0,0 +1,22 @@
+# The MIT License (MIT)
+# Copyright (c) 2024 Microsoft Corporation
+
+from azure.core.utils import CaseInsensitiveDict
+
+
+class CosmosResponse(dict):
+    def __init__(self, original_dict, response_headers):
+        super().__init__(original_dict)
+        self._response_headers = response_headers
+
+    def response_headers(self) -> CaseInsensitiveDict:
+        """Returns the response headers associated to this result
+
+        :return: Dict of response headers
+        :rtype: ~requests.structures.CaseInsensitiveDict"
+        """
+        return self._response_headers
+
+    def __repr__(self):
+        # Custom representation to include headers in the output
+        return f"{super().__repr__()} with response headers: {self._response_headers}"
```

Comment:
I would avoid adding the headers here. It could both add a lot of useless noise to customers logging/printing, it also has the potential to create a string so long that it could break things with max line lengths.